### ISSUE TITLE
[Feature]: Self-Improving Pattern Learning System

### ISSUE THREAD

**Issue by kesslerio** [kesslerio/vibe-check-mcp#132]
## Problem Statement

Currently, our pattern detection relies on static regex patterns and weights that are manually updated. While we have 87.5% accuracy, the system doesn't learn or improve from:
- New patterns encountered in the wild
- False positives/negatives reported by users
- Evolving anti-pattern variations
- Community feedback and corrections

## Proposed Solution

Implement a self-improving pattern learning system that:

1. **Tracks pattern detection outcomes**
   - Log detected patterns with confidence scores
   - Record user feedback on accuracy
   - Store pattern variations and edge cases

2. **Learns from corrections**
   - Allow users to report false positives/negatives
   - Adjust pattern weights based on feedback
   - Identify new pattern indicators

3. **Evolves pattern definitions**
   - Discover new anti-pattern variations
   - Refine existing pattern regex
   - Update thresholds based on real-world data

### Phase 1: Feedback Collection
```python
@server.tool()
def report_pattern_feedback(
    pattern_type: str,
    text_sample: str,
    was_correct: bool,
    user_explanation: Optional[str] = None
) -> Dict[str, Any]:
    """Report whether a pattern detection was accurate."""
    # Store feedback for analysis
    # Update pattern statistics
```

### Phase 2: Pattern Analytics
- Track pattern detection statistics
- Identify patterns with low confidence or high error rates
- Generate reports on pattern effectiveness

### Phase 3: Adaptive Learning
- Implement weight adjustment algorithms
- Create pattern evolution mechanisms
- Add A/B testing for pattern variations

### Phase 4: Community Learning
- Optional: Share anonymized pattern data
- Learn from aggregate community feedback
- Distribute improved patterns

## Benefits

1. **Improved Accuracy**: Evolve beyond 87.5% through continuous learning
2. **Adaptability**: Catch new anti-patterns as they emerge
3. **Reduced Maintenance**: Less manual pattern updates needed
4. **Community Value**: Collective learning benefits all users

## Privacy Considerations

- All learning data stored locally by default
- Opt-in community sharing with anonymization
- No code or sensitive data in learning records
- User control over feedback data

## Acceptance Criteria

- [ ] Feedback collection mechanism implemented
- [ ] Local pattern statistics tracking
- [ ] Weight adjustment algorithm tested
- [ ] Pattern effectiveness dashboard
- [ ] Documentation for feedback system
- [ ] Privacy controls implemented

## Technical Requirements

- Storage backend for pattern feedback (SQLite?)
- Statistical analysis module
- Pattern weight optimizer
- Feedback API endpoints

## Priority

P3 - Medium-term enhancement for continuous improvement

## Related Issues

- Pattern detection accuracy improvements
- Community feedback integration

**Comment by kesslerio** [comment]
@claude thoughts? feedback?
